source/pool/contracts/Pool.sol

```diff
@@ -252,37 +246,43 @@ contract Pool is IPool, Ownable2Step {
     uint256 _minimumAmount
   ) internal returns (uint256, bytes32[] memory) {
     if (_claims.length <= 0) revert ClaimsNotProvided();
-    uint256 _totalScore = 0;
+    uint256 _totalValue = 0;
     bytes32[] memory _rootList = new bytes32[](_claims.length);
     Claim memory _claim;
     for (uint256 i = 0; i < _claims.length; i++) {
       _claim = _claims[i];
-      if (!roots[_claim.root]) revert RootDisabled(_claim.root);
-      if (claimed[_claim.root][msg.sender]) revert AlreadyClaimed();
-      if (!verify(msg.sender, _claim.root, _claim.score, _claim.proof))
-        revert ProofInvalid(_claim.root);
-      _totalScore = _totalScore + _claim.score;
-      claimed[_claim.root][msg.sender] = true;
-      _rootList[i] = _claim.root;
+      if (rootsByTree[_claim.tree] == 0) revert TreeDisabled(_claim.tree);
+      if (claimed[_claim.tree][msg.sender]) revert AlreadyClaimed();
```

Comment:
As previously mentioned it saves gas in this function to store `rootsByTree[_claim.tree]` in an intermediate variable at the start of this loop as we check it several times..
Any gas savings are important in this function for small stakers.

@dmosites also mentioned iterating in reverse and linked [this resource](https://hackmd.io/@totomanov/gas-optimization-loops).